src/DOM/Classy/EventTarget.purs

```diff
@@ -0,0 +1,369 @@
+module DOM.Classy.EventTarget
+  ( class EventTarget
+  , addEventListener
+  , removeEventListener
+  , dispatchEvent
+  ) where
+
+import Prelude
+
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Exception (EXCEPTION)
+import DOM (DOM)
+import DOM.Classy.Event (class IsEvent, toEvent)
+import DOM.Event.EventTarget as E
+import DOM.Event.Types (EventType(..))
+import DOM.HTML.Types as H
+import DOM.Node.Types as N
+import Unsafe.Coerce (unsafeCoerce)
+
+
+class EventTarget t where
+  addEventListener :: forall eff. String -> E.EventListener (dom :: DOM | eff) -> Boolean -> t -> Eff (dom :: DOM | eff) Unit
+  removeEventListener :: forall eff. String -> E.EventListener (dom :: DOM | eff) -> Boolean -> t -> Eff (dom :: DOM | eff) Unit
```

Comment:
Can we keep `EventType` rather than `String`s here? Apart from that, looks great!